### PR TITLE
FileZilla: actually fix it for old systems

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Ok, now with wxGTK-3.0-cxx11 merged, I can fix this.
(Earlier it was building but not launching.)

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

macOS 14.4.1
Xcode 15.3

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
image

**Comment by barracuda156** [comment]
@herbygillot Could you help with merging this one, please?

File: www/FileZilla/Portfile [hunk 1]
```diff
@@ -25,13 +30,41 @@ checksums           rmd160  950e44a39b4464cd47888295eb20cb394b8df6fd \
...
+                    patch-fix-fz_datadir.diff
+}
+
+if {[string match *gcc* ${configure.compiler}]} {
+    # Yes, this is gcc-specific, or at least configure thinks so.
+    # configure: error: Boost Regex 1.76 or higher not found.
+    PortGroup       boost 1.0
+
+    boost.version   1.81
+}
```
Comment (reneeotten) [inline www/FileZilla/Portfile:58]: this is weird... is boost needed or not? That should not be specific to whether gcc or clang is used. It seems to me this needs a bit more work to sort out properly.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: www/FileZilla/Portfile (hunk 1 above)
Comment (barracuda156) [reply www/FileZilla/Portfile:58]: The build system wants to use boost-regex if GCC is used. Configure fails, not being able to find Boost (without PG it is not found). Minimum required is 1.76, but I see no reason not to use a newer one.

Comment (reneeotten) [reply www/FileZilla/Portfile:58]: but does it actually link to it, and, if so, doesn't that happen when using Clang (perhaps the it is just found correctly). In other words, you build it with Clang in trace-mode? It seems very unlikely that this is a GCC only issue..

Comment (barracuda156) [reply www/FileZilla/Portfile:58]: @reneeotten Perhaps it is a build-time dependency, and you are right to raise this issue. It should be set to build_dep only. I will sort this now.

Comment (barracuda156) [reply www/FileZilla/Portfile:58]: @reneeotten A little delay, I am trying to update it to the current version now. If that works, I will make an update here. Boost will be taken care of anyway.

Comment (barracuda156) [reply www/FileZilla/Portfile:59]: Fixed

**Comment by barracuda156** [comment]
And again CI are forcing tests when port does not have them enabled. When this finally gonna be fixed.

```
--->  Testing libfilezilla
  DEBUG: Executing proc-pre-org.macports.test-test-0
  DEBUG: system -W /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_libfilezilla/libfilezilla/work/libfilezilla-0.47.0/tests: /usr/bin/make test
  /usr/bin/clang++ -std=gnu++11 -std=c++17 -DHAVE_CONFIG_H     -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk  -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -arch arm64 -Wall -g -Werror=partial-availability -MT test-test.o -MD -MP -MF .deps/test-test.Tpo -c -o test-test.o `test -f 'test.cpp' || echo './'`test.cpp
  test.cpp:2:10: fatal error: 'cppunit/extensions/TestFactoryRegistry.h' file not found
  #include <cppunit/extensions/TestFactoryRegistry.h>
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
  make: *** [test-test.o] Error 1
  Command failed: /usr/bin/make test
```

**Comment by barracuda156** [comment]
@reneeotten Ok, looks like everything is fixed now, including tests.

**Comment by barracuda156** [comment]
@reneeotten @pmetzger Could we please merge this now?

**Pull request merged**